interfaces/builtin/greengrass_support.go

```diff
@@ -126,31 +243,23 @@ mount options=(rw, bind) /proc/irq/ -> /proc/irq/,
 mount options=(rw, bind) /proc/sys/ -> /proc/sys/,
 mount options=(rw, bind) /proc/sysrq-trigger -> /proc/sysrq-trigger,
 
-# remap a few /proc accesses to /dev/null
+# mount some devices using /dev/null
 mount options=(rw, bind) /dev/null -> /proc/kcore,
 mount options=(rw, bind) /dev/null -> /proc/sched_debug,
 mount options=(rw, bind) /dev/null -> /proc/timer_stats,
 
-# perform the pivot_root into the overlay
-pivot_root oldroot=/var/snap/greengrass/@{SNAP_REVISION}/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
-mount options=(rw, rprivate) -> /.pivot_root*/,
-umount /.pivot_root*/,
-owner /.pivot_root*/ w,
-mount options=(rw, rprivate) -> /,
-mount options=(ro, remount, rbind) -> /,
+# umounts for tearing down containers
+umount /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/*/**,
+umount /ggc-writable/packages/1.6.0/rootfs_nosys/,
+umount /ggc-writable/packages/1.6.0/rootfs_sys/,
+umount /ggc-writable/packages/1.6.0/rootfs_sys/sys/,
 
-# allow tearing down the overlay
-umount /var/snap/@{SNAP_NAME}/**,
-/run/mount/utab rw,
-/bin/umount ixr,
+# this is for container device creation
+# also need mknod and mknodat in seccomp
+capability mknod,
 
-# For lambda functions, post pivot_root lambda execution accesses
-/certs/ r,
-/certs/** r,
-/group/ r,
-/group/** r,
-/state/ r,
-/state/sqlite* rwk,
+# for the greengrassd pid file
+owner /{var/,}run/greengrassd.pid rw,
```

Comment:
The pid file is created by greengrassd which runs inside of a mount namespace we setup for it and pivot_root into, so this access is also in the post-pivot_root area. Shall I add a comment to reflect that here? Or perhaps more generally speaking I should move all accesses that are needed post-pivot_root to a specific section that is labeled clearly as being for post-pivot_root.

But to answer your question, no this can't be solved by layouts.

Reply:
I like the idea of moving the things that we know for sure are post-pivot_root somewhere-- it will be easier to audit and you can have a single comment talking about them.

Reply:
I have reorganized all the accesses to be at the end of the profile